eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/PostRestartHtlcCleaner.scala

```diff
@@ -193,24 +206,27 @@ class PostRestartHtlcCleaner(nodeParams: NodeParams, register: ActorRef, initial
           // If we haven't already settled upstream, we can fail now.
           if (!brokenHtlcs.settledUpstream.contains(origin)) {
             origin match {
-              case Origin.Local(id, _) => nodeParams.db.payments.getOutgoingPayment(id).foreach(p => {
+              case Origin.LocalCold(id) => nodeParams.db.payments.getOutgoingPayment(id).foreach(p => {
                 val payments = nodeParams.db.payments.listOutgoingPayments(p.parentId)
                 if (payments.forall(_.status.isInstanceOf[OutgoingPaymentStatus.Failed])) {
                   log.warning(s"payment failed for paymentHash=${failedHtlc.paymentHash}")
                   context.system.eventStream.publish(PaymentFailed(p.parentId, failedHtlc.paymentHash, Nil))
                 }
               })
-              case Origin.TrampolineRelayed(origins, _) =>
-                log.warning(s"payment failed for paymentHash=${failedHtlc.paymentHash}: failing ${origins.length} upstream HTLCs")
+              case Origin.ChannelRelayedCold(originChannelId, originHtlcId, _, _) =>
+                log.warning(s"payment failed for paymentHash=${failedHtlc.paymentHash}: failing 1 HTLC upstream")
+                Metrics.Resolved.withTag(Tags.Success, value = false).withTag(Metrics.Relayed, value = true).increment()
+                // We don't bother decrypting the downstream failure to forward a more meaningful error upstream, it's
+                // very likely that it won't be actionable anyway because of our node restart.
```

Comment:
Actually that's not true for channel relay.
I think we should pipe the `HtlcResult.Fail` and pattern match on it to forward the error onion upstream when available.

Reply:
Done with 6be5ba134.